cryptotest/tests/KeyFactoryTests.java

```diff
@@ -150,6 +150,15 @@ protected void checkAlgorithm(Provider.Service service, String alias) throws Alg
                     privateKeySpec = keyFactory.getKeySpec(kp.getPrivate(), PKCS8EncodedKeySpec.class);
                     publicKeySpec = keyFactory.getKeySpec(kp.getPublic(), X509EncodedKeySpec.class);
                 }
+            } else if (service.getAlgorithm().startsWith("ML-")) {
+                KeyPairGenerator kpg = KeysNaiveGenerator.getKeyPairGenerator(service.getAlgorithm(), p);
```

Comment:
this seems like dup...trip?.. quadro... duplicated chunk

Reply:
true, I can probably merge them :)

Reply:
please do!

Reply:
done